### PR TITLE
Create login form

### PR DESCRIPTION
**Pull request by jglobe**
Created button & input. Created Homepage, SignIn and SignUp pages. Added Router and simple coocke for checking redirects.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </div>
+    <BrowserRouter>
```
Comment (jqueryisamonad) [inline src/App.tsx]: Let's rewrite it to new recommended way to define routes https://reactrouter.com/en/main/routers/create-browser-router

File: package.json [hunk 2]
```diff
@@ -9,14 +9,18 @@
...
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "react-router": "^6.6.2",
+    "react-router-dom": "^6.6.2"
   },
   "devDependencies": {
     "@types/react": "^18.0.26",
     "@types/react-dom": "^18.0.9",
     "@vitejs/plugin-react": "^3.0.0",
+    "prettier": "2.8.2",
```
Comment (jqueryisamonad) [inline package.json]: remove if not using

Comment (jqueryisamonad) [reply package.json]: I will create separate task for linting

File: readme.md [hunk 3]
```diff
@@ -0,0 +1 @@
+### some-service-web
```
Comment (jqueryisamonad) [inline readme.md]:
```suggestion
# some-service-web
```

Comment (jqueryisamonad) [reply readme.md]: rename file please `README.md`

Comment (jqueryisamonad) [reply readme.md]: let's think of new project name. after we need to rename repo and change title in readme

File: src/App.css [hunk 4]
```diff
@@ -1,41 +1,39 @@
...
+	height: 100%;
 }
 
-@media (prefers-reduced-motion: no-preference) {
-  a:nth-of-type(2) .logo {
-    animation: logo-spin infinite 20s linear;
-  }
-}
+body {
+  font-family: 'Roboto', sans-serif;
```
Comment (jqueryisamonad) [inline src/App.css]: let's use `Inter` font https://rsms.me/inter/

File: src/App.css [hunk 5]
```diff
@@ -1,41 +1,39 @@
...
 
-@keyframes logo-spin {
-  from {
-    transform: rotate(0deg);
-  }
-  to {
-    transform: rotate(360deg);
-  }
+html, body {
+	height: 100%;
```
Comment (jqueryisamonad) [inline src/App.css]: I think we can remove it, check please

File: src/App.css [hunk 6]
```diff
@@ -1,41 +1,39 @@
...
-  will-change: filter;
-}
-.logo:hover {
-  filter: drop-shadow(0 0 2em #646cffaa);
-}
-.logo.react:hover {
-  filter: drop-shadow(0 0 2em #61dafbaa);
+:root {
+	--color-primary: #99c1b9;
+	--color-secondary: #8e7dbe;
```
Comment (jqueryisamonad) [inline src/App.css]: I don't understand where I should use this color, I think we should rename this variable

File: src/App.css [hunk 7]
```diff
@@ -1,41 +1,39 @@
...
 
-.card {
-  padding: 2em;
+  background-color: var(--color-background);
 }
 
-.read-the-docs {
-  color: #888;
-}
+#root {
```
Comment (jqueryisamonad) [inline src/App.css]: we can remove it. if you need page on full height just use `height: 100vh;`

File: src/App.tsx [hunk 8]
```diff
@@ -1,34 +1,31 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
+import { Routes, Route, BrowserRouter } from "react-router-dom";
+
+import { Home } from './pages/Home';
+import { SignIn, SignUp } from './pages/SignPages';
+
 import './App.css'
```
Comment (jqueryisamonad) [inline src/App.tsx]:
```suggestion
import './app.css'
```

File: src/components/Button/Button.module.css [hunk 9]
```diff
@@ -0,0 +1,27 @@
+.button_button {
```
Comment (jqueryisamonad) [inline src/components/Button/Button.module.css]: .button {}
.button_primary {}
.button_secondary {}

.button__icon {}

Comment (jqueryisamonad) [reply src/components/Button/Button.module.css]: https://ru.bem.info/methodology/

File: src/components/Button/Button.module.css [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  padding: 8px 16px;
+
+  opacity: 1;
```
Comment (jqueryisamonad) [inline src/components/Button/Button.module.css]: don't wite redundant styles

File: src/components/Input/Input.module.css [hunk 11]
```diff
@@ -0,0 +1,19 @@
+.input_input {
```
Comment (jqueryisamonad) [inline src/components/Input/Input.module.css]: https://ru.bem.info/methodology/

File: src/components/Input/Input.tsx [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  children: string;
+}
+
+export function Input({children, ...attr}: InputProps) {
+  return (
+    <input 
+      {...attr}
+      placeholder={children}
```
Comment (jqueryisamonad) [inline src/components/Input/Input.tsx]:
```suggestion
      placeholder={placeholder}
```

File: src/pages/Home/Home.module.css [hunk 13]
```diff
@@ -0,0 +1,25 @@
+.page {
+  padding: 32px;
+
+  height: 100%;
+  width: 100%;
+
+  background-color: rgba(0, 0, 0, 0.8);
+}
+
+.page_container {
```
Comment (jqueryisamonad) [inline src/pages/Home/Home.module.css]: https://ru.bem.info/methodology/

File: src/pages/Home/Home.module.css [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+  padding: 32px;
+
+  height: 100%;
+
+  border-radius: 8px;
+  background-color: var(--color-white);
+}
+
+.page_title {
```
Comment (jqueryisamonad) [inline src/pages/Home/Home.module.css]: let's move it to separate component `<PageTitle>`

File: src/pages/Home/Home.tsx [hunk 15]
```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { Button } from '../../components/Button';
```
Comment (jqueryisamonad) [inline src/pages/Home/Home.tsx]: let's setup aliased imports:
```suggestion
import { Button } from '@/components/Button';
```

https://stackoverflow.com/questions/43281741/how-to-use-paths-in-tsconfig-json
https://koenwoortman.com/vitejs-module-aliases/

File: src/pages/SignPages/Sign.module.css [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+  top: 0;
+  left: 0;
+
+  height: 100%;
+  width: 100%;
+
+  background-color: rgba(0, 0, 0, 0.8);
+}
+
+.container_inner {
```
Comment (jqueryisamonad) [inline src/pages/SignPages/Sign.module.css]: https://ru.bem.info/methodology/

File: src/pages/SignPages/Sign.module.css [hunk 17]
```diff
@@ -0,0 +1,53 @@
+.container {
+  position: fixed;
```
Comment (jqueryisamonad) [inline src/pages/SignPages/Sign.module.css]: I think we shoudn't use fixed position for the whole page

File: src/pages/SignPages/Sign.module.css [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+  left: 0;
+
+  height: 100%;
+  width: 100%;
+
+  background-color: rgba(0, 0, 0, 0.8);
+}
+
+.container_inner {
+  position: absolute;
```
Comment (jqueryisamonad) [inline src/pages/SignPages/Sign.module.css]: It's unneccesary. `display: flex` for container is enough

File: src/pages/SignPages/Sign.module.css [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+  line-height: 1.5;
+}
+
+.form_error {
+  height: 20px;
+
+  color: var(--color-error);
+}
+
+.form_link {
```
Comment (jqueryisamonad) [inline src/pages/SignPages/Sign.module.css]: let's move it to separate component <Link>

File: src/pages/SignPages/SignIn.tsx [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+    <div className={styles.container}>
+      <div className={styles.container_inner}>
+        <form
+          className={styles.form}
+          onSubmit={login}
+        >
+          <h2 className={styles.form_title}>
+            Sign In
+          </h2>
+          <Input
```
Comment (jqueryisamonad) [inline src/pages/SignPages/SignIn.tsx]: Add labels for inputs `<Label>`

File: src/pages/SignPages/SignIn.tsx [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+            priority='primary'
+            type='submit'
+          >
+            Sign In
+          </Button>
+          <a 
+            href="/signup"
+            className={styles.form_link}
+          >
+              Create new accaunt
```
Comment (jqueryisamonad) [inline src/pages/SignPages/SignIn.tsx]:
```suggestion
              Create new account
```

File: src/App.css [hunk 22]
```diff
@@ -1,41 +1,39 @@
...
-.logo.react:hover {
-  filter: drop-shadow(0 0 2em #61dafbaa);
+:root {
+	--color-primary: #99c1b9;
+	--color-secondary: #8e7dbe;
+	--color-tertiary: #f2d0a9;
+  --color-error: #d88c9a;
+	--color-font-primary: #000000;
+	--color-font-secondary: #909090;
+	--color-background: #ffffff;
```
Comment (jqueryisamonad) [inline src/App.css]: if you have two backgrounds in the app consider to create `--color-background-primary` and `--color-background-secondary`

`--color-background-primary` — main app background
`--color-background-secondary` — background for some block on the page, e.g. sign in form

File: src/components/Button/Button.tsx [hunk 23]
```diff
@@ -0,0 +1,24 @@
+import cn from 'classnames';
+
+import styles from './Button.module.css';
+
+interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  priority: string;
+  children: React.ReactNode;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
```
Comment (jqueryisamonad) [inline src/components/Button/Button.tsx]: `React.ButtonHTMLAttributes<HTMLButtonElement>` includes types for `children` and `onClick`. There is no need to define these types again

File: src/components/Input/Input.module.css [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+  padding: 4px;
+
+  height: 34px;
+  border: solid 2px var(--color-font-secondary);
```
Comment (jqueryisamonad) [inline src/components/Input/Input.module.css]: border has color of font? looks like error

File: src/components/Input/Input.module.css [hunk 25]
```diff
@@ -0,0 +1,19 @@
...
+  border-radius: 4px;
+
+  color: var(--color-font-secondary);
```
Comment (jqueryisamonad) [inline src/components/Input/Input.module.css]: Entered text looks almost like placeholder. Make it more visible (darker)

image

File: src/components/Button/Button.tsx [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export function Button({priority, children, onClick, ...attr}: ButtonProps) {
+  return (
+    <button
+      {...attr}
+      className={cn({
+        [styles[`${'button_button__' + priority}`]]: priority,
```
Comment (jqueryisamonad) [inline src/components/Button/Button.tsx]: template literals can sometimes be unnecessary

```suggestion
        [styles['button_button__' + priority]]: priority,
```

File: src/pages/Home/Home.tsx (hunk 15 above)
Comment (jqueryisamonad) [reply src/pages/Home/Home.tsx]: please create separate PR for this

**Review by jqueryisamonad — CHANGES_REQUESTED** [review]
Great work! First PRs always have a lot of to do. It gets easier from here!

File: src/app.css [hunk 27]
```diff
@@ -0,0 +1,46 @@
...
+
+	font-family: 'Inter', sans-serif;
+}
+
+@supports (font-variation-settings: normal) {
+  :root { font-family: 'Inter var', sans-serif; }
+}
+
+body {
+	height: 100vh;
```
Comment (jqueryisamonad) [inline src/app.css]: There is no need to define the height for the body, use it only on pages where it is needed.

File: src/app.css [hunk 28]
```diff
@@ -0,0 +1,46 @@
...
+
+  font-family: 'Inter', sans-serif;
+  font-size: 16px;
+  line-height: 1.2;
+  color: var(--color-font-primary);
+
+  background-color: var(--color-background-primary);
+}
+
+#root {
```
Comment (jqueryisamonad) [inline src/app.css]: No need to define the height for root

File: src/app.tsx [hunk 29]
```diff
@@ -0,0 +1,32 @@
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+
+import { Home } from './pages/home';
+import { SignIn, SignUp } from './pages/sign-pages';
```
Comment (jqueryisamonad) [inline src/app.tsx]:
```suggestion
import { SignIn, SignUp } from './pages/auth';
```

File: src/components/button/button.module.css [hunk 30]
```diff
@@ -0,0 +1,17 @@
...
+  
+  cursor: pointer;
+}
+
+.button:hover {
+  opacity: .8;
+}
+
+.button_primary {
+  color: var(--color-white);
```
Comment (jqueryisamonad) [inline src/components/button/button.module.css:15]: bad contrast ration

image

File: src/components/link/link.module.css [hunk 31]
```diff
@@ -0,0 +1,8 @@
+.link {
+  color: var(--color-link);
```
Comment (jqueryisamonad) [inline src/components/link/link.module.css:2]: bad contrast ration

image

File: src/components/input/input.module.css [hunk 32]
```diff
@@ -0,0 +1,42 @@
+.label {
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 4px;
+
+  font-size: 16px;
+  color: var(--color-font-secondary);
```
Comment (jqueryisamonad) [inline src/components/input/input.module.css:8]: bad contrast ration

image

File: src/components/input/input.module.css [hunk 33]
```diff
@@ -0,0 +1,42 @@
+.label {
```
Comment (jqueryisamonad) [inline src/components/input/input.module.css]: You have an `input` file, but the main block is `label`. Is this correct?

File: src/components/input/input.module.css [hunk 34]
```diff
@@ -0,0 +1,42 @@
...
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 4px;
+
+  font-size: 16px;
+  color: var(--color-font-secondary);
+}
+
+.label:hover  {
+  color: var(--color-primary);
```
Comment (jqueryisamonad) [inline src/components/input/input.module.css]: Have you often seen the label color change when hovering over other sites?

File: src/components/link/link.tsx [hunk 35]
```diff
@@ -0,0 +1,19 @@
+import { Link as RouterLink } from "react-router-dom";
+
+import styles from './link.module.css';
+
+interface LinkType {
+  to: string;
+  children: string;
```
Comment (jqueryisamonad) [inline src/components/link/link.tsx]: Will this type work in this case:
```jsx
<Link>
  Go to <b>login<b> page
</Link>
```

File: src/components/page-title/page-title.tsx [hunk 36]
```diff
@@ -0,0 +1,13 @@
+import styles from './page-title.module.css';
+
+interface PageTitleType {
+  children: string;
```
Comment (jqueryisamonad) [inline src/components/page-title/page-title.tsx]: Will this type work in this case:
```jsx
<PageTitle>
  <SettingsIcon /> Settings
</PageTitle>
```

**Review by jqueryisamonad — CHANGES_REQUESTED** [review]

File: src/pages/Home/Home.tsx (hunk 15 above)
Comment (jglobe) [reply src/pages/Home/Home.tsx]: Now i make it resolved, i will vreate new pr

File: src/components/input/input.module.css (hunk 32 above)
Comment (jglobe) [reply src/components/input/input.module.css:8]: contrast ratio > 5 is good?

**Pull request merged**